_posts/2022-11-10-why-python-package-health-metrics-matter.md

```diff
@@ -0,0 +1,111 @@
+model helps you define how 
+to collect the data that you need to track outcome success. 
+
+## Baseline data are critical to collect at the start of building a program or organization
+
+As I am building the pyOpenSci program, I find myself thinking about what metrics 
+around Python scientific open source software we want to track to better understand:
+
+1. The outcomes of our activities
+2. The overall health of packages in our growing pyOpenSci ecosystem (specific to our organization)
+3. How/if we contributed to improving the health of packages in our ecosystem
+4. How we are impacting the broader scientific python, open source community
+
+### Baseline metrics collected from the start will help your future self when running or working in an organization
+
+As mentioned above, collecting metrics from the start 
+of your efforts allows you to get off the ground running with data that you can use to compare to future data. Thus while it may not be the work that you want to do, it will 
+help your future self. 
+
+For pyOpenSci, collecting metrics allows us in the future 
+to evaluate our programs and adaptively change things to 
+make sure we are getting the outcomes that we want. 
+
+Outcomes such as 
+* Scientists being better able to find packages that are maintained to support their workflows  
+* Improved documentation in packages as a result of our reviews
+
+### But what metrics should we collect about scientific Python packages?
+
+[In a previous post,](/blog/2022-10-24-why-should-python-open-source-matter-science) I spoke 
+generally about why open source should matter to you as a scientist and as a 
+developer or package maintainer. 
+
+To better understand what data we should be collecting to track our packages 
+health over time, I went to Twitter to see what my colleagues around the world
+had to say. That conversation resulted in some really interesting insights.
+
+Most importantly, it allowed me to begin to break down and group metrics in 
+terms of pyOpenSci goals. 
+
+
+### Goals for package metrics
+
+We hope that:
+
+* Peer review improves Python package structure and usability. 
+* Peer review in some way supports maintenance and/or responsible archiving when a package comes to life-end.
+* Over time, the package is improved and maintained with possible contributions for those other than the maintainer. 
+
+We need metrics to understand things like
+
+* Community adoption of the package (are scientists using it?)
+* Maintenance level of the package (are maintainers still working on it and fixing bugs?)
+* Infrastructure (are tests setup to help identify if contributions break things? )
+* Usability (is the package documented in a way that helps users quickly get started)
+
+
+## Three Python open source software healthy metric "buckets" that pyOpenSci cares about
+
+Based on all of the feedback, and what I *think* might be a start at what pyOpenSci  
```

Comment:
Overall everything you're saying here makes perfect sense to me, but I was kind of surprised this was the end of the post. I thought you were going to elaborate more on why you chose those 3 buckets, e.g., based on comments on twitter. Right now it sounds a bit like you are repeating what you just said above, and that you are defining your metrics to be exactly the things you just said we need to understand? Maybe give a little more detail about how you would actually compute each metric here?
For example "to look at community adoption we will consider number of github stars, number of downloads and forks, and number of stackoverflow posts" (or whatever). This could be a good place to reference rOpenSci and fatiando as an illustration of how other communities have approached this?

I know you're trying to close out the post here and I'm not trying to suggest you add a ton more. Maybe merge this section and the one above it on goals? "Goals of metrics and three possible buckets"

Reply:
totally. this blog was supposed to setup the next blog and that summary at the end is supposed to kind of setup - if you want to understand where these came from - read the next blog. but it sounds like that wasn't clear in the post? 
In terms of how we calculate i was thinking that would be a 4th blog.

let me know what you think as i don't want to leave people hanging here.

Reply:
Overall the post reads clearly to me now but it still feels a bit like you repeat yourself at the end, and now that I have read the first post, it feels like you are repeating yourself from that post too

If it were me I would rewrite like so:
* in my last post I translated what scientists need from open source packages into some goals for package metrics, [here](https://www.pyopensci.org/blog/why-python-open-source-software-matters-for-scientists#goals-for-package-metrics). Let me restate those here:
  * goals, leave out the metrics since you're about to say this anyways
* We need metrics that match these goals, but as I discussed this with others I realized that we also need a holistic view of package health.
* I have chosen three dimensions (or however many you decide on) along which we can consider package health, and we can group any number of metrics along these dimensions
  * maintenance
  * infrastructure
  * community
In the last post in this series I will share some of the discussion on twitter that helped me come to this conclusion. I hope you'll continue the discussion here and on twitter as well.